network/ServerNetworking.cpp

```diff
@@ -30,17 +30,24 @@ namespace {
 class DiscoveryServer {
 public:
     DiscoveryServer(boost::asio::io_context& io_context) :
-        m_socket(io_context, udp::endpoint(udp::v6(), Networking::DiscoveryPort()))
+        m_service(io_context),
+        m_socket(io_context)
     { Listen(); }
 
 private:
     void Listen() {
-        m_recv_buffer.fill('\0');
-        m_socket.async_receive_from(
-            boost::asio::buffer(m_recv_buffer),
-            m_remote_endpoint,
-            boost::bind(&DiscoveryServer::HandleReceive, this,
-                        boost::asio::placeholders::error));
+        // use a dual stack (ipv6 + ipv4) socket
+        udp::endpoint discovery_endpoint(udp::v6(), Networking::DiscoveryPort());
+
+        if (!GetOptionsDB().Get<bool>("singleplayer")) {
+            m_socket = udp::socket(m_service, discovery_endpoint);
```

Comment:
Instead of checking if it's a single or multi-player game, maybe put a try-catch block around this, or whichever line throws the exception that was causing the issue? Only setting up the socket in MP won't actually help if the MP server only have IPV4, but try-catch would avoid a crash in either case. Could fall-back to an IPV4 endpoint as well, if setting up the IPV6 throws.

Reply:
Added try-catch block both for message and discovery ports.